### PR TITLE
KEYCLOAK-19316 Add Warning about 2nd-factor authenticator setup

### PR DESCRIPTION
**Pull request by mposolda**

### CODE REVIEW

**Comment by mposolda** [comment]
@andymunro Do you please have a chance to review this?

File: server_admin/topics/authentication/flows.adoc [hunk 1]
```diff
@@ -132,6 +132,17 @@ Executions have a *Config* menu option to configure the execution. It is also po
 
 Since the order of execution is important, you can move executions and sub-flows up and down within their flows using the up and down buttons beside their names.
 
+[WARNING]
+====
+Make sure to properly test your configuration when you configure the authentication flow to doublecheck there are no security holes in your setup. It is recommended to test various
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: ..to confirm that no security holes exist in your setup. We recommend that you test various
corner cases.

File: server_admin/topics/authentication/flows.adoc [hunk 2]
```diff
@@ -132,6 +132,17 @@ Executions have a *Config* menu option to configure the execution. It is also po
...
+corner cases. Like for example the behaviour of authentication for the particular user when you remove various credentials from his account before the authentication.
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: For example, consider testing the authentication behavior for a user when you remove various credentials from the user's account before authentication.

File: server_admin/topics/authentication/flows.adoc [hunk 3]
```diff
@@ -132,6 +132,17 @@ Executions have a *Config* menu option to configure the execution. It is also po
...
+
+As an example, when 2nd-factor authenticators, like OTP Form or WebAuthn Authenticator, are configured in the flow as REQUIRED and user does not have credential of particular
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: generally replace "like" with "such as" or "for example"
add "the" before "user"

File: server_admin/topics/authentication/flows.adoc [hunk 4]
```diff
@@ -132,6 +132,17 @@ Executions have a *Config* menu option to configure the execution. It is also po
...
+type, the user will be able to setup the particular credential during authentication itself. This means that in fact he does not authenticate with this credential as he setup it
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: Always follow "This" with a noun.  "This situation..."
Remove "in fact" which adds no meaning.
Replace "he" with "the user" to make it gender-neutral

File: server_admin/topics/authentication/flows.adoc [hunk 5]
```diff
@@ -132,6 +132,17 @@ Executions have a *Config* menu option to configure the execution. It is also po
...
+right during the authentication. So for browser authentication, make sure to configure your authentication flow with some 1st-factor credentials like Password or WebAuthn
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: replace "like" with "such as"
Remove "So" which adds no meaning

**Review by andymunro — CHANGES_REQUESTED** [review]
Looks good, @mposolda.  Just a few changes needed.

**Comment by mposolda** [comment]
@andymunro Thanks for the review! I've included all your suggestions.
